### PR TITLE
examples/window: add macOS and Windows window example

### PR DESCRIPTION
**Pull request by TotallyGamerJet**

### CODE REVIEW

File: examples/window/main_darwin.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+func init() {
+	runtime.LockOSThread()
+}
+
+func main() {
+	if _, err := purego.Dlopen("/System/Library/Frameworks/Cocoa.framework/Cocoa", purego.RTLD_GLOBAL|purego.RTLD_LAZY); err != nil {
+		panic(err)
+	}
+	NSApp := objc.ID(objc.GetClass("NSApplication")).Send(objc.RegisterName("sharedApplication"))
```
Comment (hajimehoshi) [inline examples/window/main_darwin.go:40]: `nsApp`

File: examples/window/main_windows.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024 The Ebitengine Authors
+
+package main
+
+import (
+	"unsafe"
+
+	"github.com/ebitengine/purego"
+	"golang.org/x/sys/windows"
```
Comment (hajimehoshi) [inline examples/window/main_windows.go:10]: Let's reorder them like this:

```go
import (
    "unsafe"

    "golang.org/x/sys/windows"

    "github.com/ebitengine/purego"
)
```

File: examples/window/main_windows.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+		CW_USEDEFAULT, CW_USEDEFAULT, int(wr.Right-wr.Left), int(wr.Bottom-wr.Top),
+		0, 0, inst, nil,
+	)
+	if hwnd == 0 {
+		panic(windows.GetLastError())
+	}
+
+	ShowWindow(hwnd, SW_SHOW)
+
+	msg := MSG{}
```
Comment (hajimehoshi) [inline examples/window/main_windows.go:122]:
```go
var msg MSG
```

File: examples/window/main_windows.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+	wc := WNDCLASSEX{
+		Size:      uint32(unsafe.Sizeof(WNDCLASSEX{})),
+		WndProc:   windows.NewCallback(wndProc),
+		Instance:  inst,
+		ClassName: windows.StringToUTF16Ptr(className),
+	}
+
+	RegisterClassEx(&wc)
+
+	wr := RECT{0, 0, 320, 240}
```
Comment (hajimehoshi) [inline examples/window/main_windows.go:107]:
```go
wr := RECT{
    Left: 0,
    Top: 0,
    Right: 320,
    Bottom: 240,
}
```

File: examples/window/main_darwin.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	Width, Height float64
+}
+
+type NSRect struct {
+	Origin NSPoint
+	Size   NSSize
+}
+
+func init() {
+	runtime.LockOSThread()
```
Comment (hajimehoshi) [inline examples/window/main_darwin.go:33]: Don't we need the same thing for Windows?

**Comment by hajimehoshi** [comment]
Please let me know when this is ready, thanks!

**Comment by TotallyGamerJet** [comment]
It's ready

**Pull request merged**